src/SFA.DAS.ApprenticeFeedback.Jobs/Functions/GenerateEmailTransactionsFunction.cs

```diff
@@ -0,0 +1,43 @@
+﻿
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using SFA.DAS.ApprenticeCommitments.Jobs.Api;
+using SFA.DAS.ApprenticeFeedback.Jobs.Infrastructure.Api.Requests;
+
+
+namespace SFA.DAS.ApprenticeFeedback.Jobs.Functions
+{
+    public class GenerateEmailTransactionsFunction
+    {
+        private readonly ILogger<GenerateEmailTransactionsFunction> _log;
+        private readonly IApprenticeFeedbackEmailTransactionApi _api;
+
+        public GenerateEmailTransactionsFunction(ILogger<GenerateEmailTransactionsFunction> log, IApprenticeFeedbackEmailTransactionApi api) //, IApprenticeFeedbackEmailTransactionService service)
+        {
+            _api = api;
+            _log = log;
+        }
+
+        [FunctionName("GenerateEmailTransactions")]
+        public async Task Run([TimerTrigger("%FunctionsOptions:GenerateEmailTransactionsOptions:Schedule%", RunOnStartup = false)] TimerInfo myTimer,
```

Comment:
You may or may not want to take the logic out of the function and transfer to Nservice bus like [this](https://github.com/SkillsFundingAgency/das-apprentice-feedback-jobs/blob/main/src/SFA.DAS.ApprenticeFeedback.Jobs/Functions/ApprenticeFeedbackSummaryTimerTrigger.cs#L17)

depending on if you think there is value in testing the event like [this](https://github.com/SkillsFundingAgency/das-apprentice-feedback-jobs/blob/main/src/SFA.DAS.ApprenticeFeedback.Jobs.UnitTests/WhenGeneratingApprenticeFeedbackSummaries.cs)

Reply:
Not sure what you mean by this - what logic?